tools/parquetDDL/main.cpp

```diff
@@ -0,0 +1,285 @@
+*/
+void getSchema(std::string filePath, std::shared_ptr<arrow::Schema>* parquetSchema)
+{
+  std::shared_ptr<arrow::io::ReadableFile> infile;
+  PARQUET_ASSIGN_OR_THROW(infile, arrow::io::ReadableFile::Open(filePath, arrow::default_memory_pool()));
+  std::unique_ptr<parquet::arrow::FileReader> reader;
+  PARQUET_THROW_NOT_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+  PARQUET_THROW_NOT_OK(reader->GetSchema(parquetSchema));
+  PARQUET_THROW_NOT_OK(infile->Close());
+}
+
+/**
+ * convert arrow data type id to corresponding columnstore type string
+*/
+int convert2mcs(std::shared_ptr<arrow::DataType> dataType, arrow::Type::type typeId, std::string& colType)
+{
+  switch (typeId)
+  {
+    case arrow::Type::type::BOOL:
+    {
+      colType = "BOOLEAN";
+      break;
+    }
+    case arrow::Type::type::UINT8:
+    {
+      colType = "TINYINT UNSIGNED";
+      break;
+    }
+    case arrow::Type::type::INT8:
+    {
+      colType = "TINYINT";
+      break;
+    }
+    case arrow::Type::type::UINT16:
+    {
+      colType = "SMALLINT UNSIGNED";
+      break;
+    }
+    case arrow::Type::type::INT16:
+    {
+      colType = "SMALLINT";
+      break;
+    }
+    case arrow::Type::type::UINT32:
+    {
+      colType = "INT UNSIGNED";
+      break;
+    }
+    case arrow::Type::type::INT32:
+    {
+      colType = "INT";
+      break;
+    }
+    case arrow::Type::type::UINT64:
+    {
+      colType = "BIGINT UNSIGNED";
+      break;
+    }
+    case arrow::Type::type::INT64:
+    {
+      colType = "BIGINT";
+      break;
+    }
+    case arrow::Type::type::FLOAT:
+    {
+      colType = "FLOAT";
+      break;
+    }
+    case arrow::Type::type::DOUBLE:
+    {
+      colType = "DOUBLE";
+      break;
+    }
+    case arrow::Type::type::STRING:
+    {
+      // set 2000 as the maximum length and VARCHAR as column type
+      colType = "VARCHAR(2000)";
+      break;
+    }
+    case arrow::Type::type::BINARY:
+    {
+      // set 8000 as the maximum length and VARCHAR as column type
+      colType = "VARCHAR(8000) character set 'binary'";
+      break;
+    }
+    case arrow::Type::type::FIXED_SIZE_BINARY:
+    {
+      std::shared_ptr<arrow::FixedSizeBinaryType> fType = std::static_pointer_cast<arrow::FixedSizeBinaryType>(dataType);
+      int byteWidth = fType->byte_width();
+      colType = "CHAR(" + std::to_string(byteWidth) + ")";
+      break;
+    }
+    case arrow::Type::type::DATE32:
+    {
+      colType = "DATE";
+      break;
+    }
+    case arrow::Type::type::DATE64:
+    {
+      colType = "DATE";
+      break;
+    }
+    case arrow::Type::type::TIMESTAMP:
+    {
+      std::shared_ptr<arrow::TimestampType> fType = std::static_pointer_cast<arrow::TimestampType>(dataType);
+
+      if (fType->unit() == arrow::TimeUnit::MILLI)
+        colType = "TIMESTAMP(3)";
+      else if (fType->unit() == arrow::TimeUnit::MICRO)
+        colType = "TIMESTAMP(6)";
+      else
+        return UNSUPPORTED_DATA_TYPE;
```

Comment:
This worth an additional printout so that it is obvious what is the unsupported element.

Reply:
Caller of this function handles this return code and provides error message.